File: src/NetFxCSharp/Facesso/ToolStripMonthCalender.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso
{
    public class ToolStripMonthCalender : ToolStripControlHost
    {
        // Call the base constructor passing in a MonthCalendar instance.
        public ToolStripMonthCalender() : base(new MonthCalendar())
        {
        }

        public MonthCalendar MonthCalendarControl
        {
            get
            {
                return ((MonthCalendar)Control);
            }
        }

        // Expose the MonthCalendar.FirstDayOfWeek as a property.
        public Day FirstDayOfWeek
        {
            get
            {
                return MonthCalendarControl.FirstDayOfWeek;
            }

            set
            {
                value = MonthCalendarControl.FirstDayOfWeek;
            }
        }

        public System.DateTime Value
        {
            get
            {
                return MonthCalendarControl.SelectionStart;
            }

            set
            {
                MonthCalendarControl.SelectionStart = value;
                MonthCalendarControl.SelectionEnd = value;
            }
        }

        // Expose the AddBoldedDate method.
        public void AddBoldedDate(DateTime dateToBold)
        {
            MonthCalendarControl.AddBoldedDate(dateToBold);
        }

        public void RemoveBoldedDate(DateTime dateToUnbold)
        {
            MonthCalendarControl.RemoveBoldedDate(dateToUnbold);
        }

        public void RemoveAllBoldedDates()
        {
            MonthCalendarControl.RemoveAllBoldedDates();
        }

        // Subscribe and unsubscribe the control events you wish to expose.
        protected override void OnSubscribeControlEvents(Control c)
        {
            // Call the base so the base events are connected.
            base.OnSubscribeControlEvents(c);
            // Cast the control to a MonthCalendar control.
            MonthCalendar monthCalendarControl = ((MonthCalendar)c);
            // Add the event.
            monthCalendarControl.DateChanged += HandleDateChanged;
        }

        protected override void OnUnsubscribeControlEvents(Control c)
        {
            // Call the base method so the basic events are unsubscribed.
            base.OnUnsubscribeControlEvents(c);
            // Cast the control to a MonthCalendar control.
            MonthCalendar monthCalendarControl = ((MonthCalendar)c);
            // Remove the event.
            monthCalendarControl.DateChanged -= HandleDateChanged;
        }

        // Declare the DateChanged event.
        public event DateRangeEventHandler DateChanged;
        // Raise the DateChanged event.
        private void HandleDateChanged(object sender, DateRangeEventArgs e)
        {
            DateChanged?.Invoke(this, e);
        }
    }
}

File: src/NetFxCSharp/Facesso/ToolStripMonthCalender.vb
Public Class ToolStripMonthCalender
    Inherits ToolStripControlHost

    ' Call the base constructor passing in a MonthCalendar instance.
    Public Sub New()
        MyBase.New(New MonthCalendar())
    End Sub

    Public ReadOnly Property MonthCalendarControl() As MonthCalendar
        Get
            Return CType(Control, MonthCalendar)
        End Get
    End Property

    ' Expose the MonthCalendar.FirstDayOfWeek as a property.
    Public Property FirstDayOfWeek() As Day
        Get
            Return MonthCalendarControl.FirstDayOfWeek
        End Get
        Set(ByVal value As Day)
            value = MonthCalendarControl.FirstDayOfWeek
        End Set
    End Property

    Public Property Value() As Date
        Get
            Return MonthCalendarControl.SelectionStart
        End Get
        Set(ByVal value As Date)
            MonthCalendarControl.SelectionStart = value
            MonthCalendarControl.SelectionEnd = value
        End Set
    End Property

    ' Expose the AddBoldedDate method.
    Public Sub AddBoldedDate(ByVal dateToBold As DateTime)
        MonthCalendarControl.AddBoldedDate(dateToBold)
    End Sub

    Public Sub RemoveBoldedDate(ByVal dateToUnbold As DateTime)
        MonthCalendarControl.RemoveBoldedDate(dateToUnbold)
    End Sub

    Public Sub RemoveAllBoldedDates()
        MonthCalendarControl.RemoveAllBoldedDates()
    End Sub

    ' Subscribe and unsubscribe the control events you wish to expose.
    Protected Overrides Sub OnSubscribeControlEvents(ByVal c As Control)

        ' Call the base so the base events are connected.
        MyBase.OnSubscribeControlEvents(c)

        ' Cast the control to a MonthCalendar control.
        Dim monthCalendarControl As MonthCalendar = _
            CType(c, MonthCalendar)

        ' Add the event.
        AddHandler monthCalendarControl.DateChanged, _
            AddressOf HandleDateChanged
    End Sub

    Protected Overrides Sub OnUnsubscribeControlEvents(ByVal c As Control)
        ' Call the base method so the basic events are unsubscribed.
        MyBase.OnUnsubscribeControlEvents(c)

        ' Cast the control to a MonthCalendar control.
        Dim monthCalendarControl As MonthCalendar = _
            CType(c, MonthCalendar)

        ' Remove the event.
        RemoveHandler monthCalendarControl.DateChanged, _
            AddressOf HandleDateChanged

    End Sub

    ' Declare the DateChanged event.
    Public Event DateChanged As DateRangeEventHandler

    ' Raise the DateChanged event.
    Private Sub HandleDateChanged(ByVal sender As Object, _
        ByVal e As DateRangeEventArgs)

        RaiseEvent DateChanged(Me, e)
    End Sub


End Class
